### PR TITLE
MINOR: update checkstyle

### PR DESCRIPTION
**Pull request by vvcephei**
Update to checkstyle 8.18

### CODE REVIEW

**Comment by vvcephei** [comment]
Thanks for the review, @ewencp . The reason we have the version here is that we needed a feature that was unavailable in the parent pom. If we updated the parent pom at that time, then other child projects may fail to build if they're incompatible with the later version. It's impractical to try and keep every project in lockstep sync on all dependencies and plugins.

Now, of course, there's a motivation to update the parent to at least 8.18, which I'm happy to also send a PR for.

Regarding the advice about using the version, I'm curious about the motivation for declaring variables that are used in exactly one place. This is a widespread and common pattern with Maven, but in my experience there are only downsides to it. Maybe your experience is different. This practice creates a management problem in combination with hierarchical overrides up the pom inheritance chain. Consider that this pom has 3 levels of parent poms, each of which can independently override the version or the configuration of the checkstyle plugin, we to understand what version and configuration we're getting, we have to look in 8 different places (actually 12, since this plugin has two different version number variables). Conversely, since we explicitly set the version number with no variables directly in our pom, we can fully understand our checkstyle usage by looking in just one place.

I can see the value of using a variable for something like AK, because there will be many dependencies that need to use the exact same version string. I'm only questioning the practice of declaring version variables that are used in only one place. My current stance is that it's reasonable to declare a default plugin configuration in the parent pom, but we should be fine with leaf projects declaring their own configurations (including the version) if they need something different from the default.

:+1: on backporting. I'll do a little archaeology and figure out the right insertion point for this change.

**Comment by vvcephei** [comment]
As requested, I've rebased the change against 3.3.0-post. If I understand the versioning correctly, this is the very first branch we're maintaining, so once I merge this change to all branches, we shouldn't have any further problems with non-final variables in older branches getting pint-merged to later branches. (In addition to making sure we're on checkstyle 8.18 in all branches).

**Comment by vvcephei** [comment]
I've gone ahead and re-requested reviews from everyone, since the scope of the change is now much larger. For reference, most of the lines of change are just adding `final` to variables. I've already done a pass in the IDEA branch comparer to make sure nothing weird happened with the formatting.

File: src/main/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesRestService.java [hunk 1]
```diff
@@ -78,13 +78,13 @@
...
 
-    HostStoreInfo hostStoreInfo = streamsMetadataForStoreAndKey(storeName, key);
+    final HostStoreInfo hostStoreInfo = streamsMetadataForStoreAndKey(storeName, key);
     if (!thisHost(hostStoreInfo)){
        return fetchByKey(hostStoreInfo, "state/keyvalue/"+storeName+"/"+key);
     }
 
     // Lookup the KeyValueStore with the provided storeName
-    final ReadOnlyKeyValueStore<String, Long> store = streams.store(storeName, QueryableStoreTypes.<String, Long>keyValueStore());
+    final ReadOnlyKeyValueStore<String, Long> store = streams.store(storeName, QueryableStoreTypes.keyValueStore());
```
[the remark this answers is not in the document]
Comment (vvcephei) [reply src/main/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesRestService.java]: I wondered about that, too, so I checked the pom. The language level is 1.8 for this project in 3.3.0. I guess, even though AK supports java 7 in 0.11.0, we still wrote the examples against java 8 (which is also supported in 0.11.0)?

[the remark this answers is not in the document]
Comment (mjsax) [reply src/main/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesRestService.java]: Yes, 0.11.0 supports Java 7 and Java 8 -- if we wrote the examples with target version 8, it's fine. Just wanted to double check.

**Comment by vvcephei** [comment]
Hey @ewencp , just a heads-up, I'd like to merge this PR after the tests pass.

The version of checkstyle available in then 3.3.0 common-parent is doesn't support requiring final, which we want to do in this repo, so I'd still like to merge this PR. Since you've updated the parent pom in trunk, after I pint-merge this changeset, I'll send a follow-up PR to trunk to remove the version override.

My motivation to go ahead and merge this PR is just that we've had some maintenance problems, as people make code changes in older branches than then don't build once they're merged to newer branches that do require final. After this PR, the requirement will be in place on every version of this repo.

**Comment by ewencp** [comment]
@vvcephei sure that makes sense. fallout of updates was not huge, but definitely widespread, so it would be a pain to update that now (tbh it seems crazy that  upgrades are changing behavior enough on what I thought was a pretty mature tool).

In terms of merge, you can also just do the merging manually and clean up at the merge to 5.0.x if you want.

**Comment by ewencp** [comment]
Oh, and btw, given the update, have you merged and checked newer branches don't require additional changes? This is one of the challenges of making changes like this on older branches -- you need to validate all to-be-merged branches in addition to the Jenkins validation you get on this branch.

**Comment by vvcephei** [comment]
Thanks @ewencp !

I fully expect to have to run the validation after every merge, but hopefully, this will save everyone else from a proportional amount of pain in the future.

**Comment by vvcephei** [comment]
Ok! The tests passed. I'll start the process of manually merging tomorrow morning, to ensure I have enough time to complete it without having to leave a 16-hour gap where it's partially merged.

**Pull request merged**